cluster/manager_common_updater.go

```diff
@@ -168,8 +174,72 @@ func (c *commonUpdater) Update(ctx context.Context) error {
 	}
 
 	// on certain clouds like Alibaba & Ec2_Banzaicloud we still need to add node pool name labels
-	if err := LabelNodesWithNodePoolName(c.cluster); err != nil {
+	if err := labelNodesWithNodePoolName(c.cluster); err != nil {
 		return emperror.Wrap(err, "adding labels to nodes failed")
 	}
 	return nil
 }
+
+// labelNodesWithNodePoolName add node pool name labels for all nodes.
+// It's used only used in case of ACK etc. when we're not able to add labels via API.
+func labelNodesWithNodePoolName(commonCluster CommonCluster) error {
+	switch commonCluster.GetDistribution() {
+	case pkgCluster.EKS, pkgCluster.OKE, pkgCluster.GKE, pkgCluster.PKE:
+		log.Infof("nodes are already labelled on : %v", commonCluster.GetDistribution())
+		return nil
+	}
+
+	log.Debug("get K8S config")
+	kubeConfig, err := commonCluster.GetK8sConfig()
+	if err != nil {
+		return err
+	}
+
+	log.Debug("get K8S connection")
+	client, err := k8sclient.NewClientFromKubeConfig(kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	nodeNameLister, ok := commonCluster.(nodeNameLister)
```

Comment:
This should be switched with Kubernetes client creation. Interface check is more lightweight, it should fail fast.